order-api/src/main/java/com/smoothstack/order/api/OrderApiController.java

```diff
@@ -43,16 +44,17 @@ public OrderApiController(NativeWebRequest request, OrderService orderService) {
 
     @Override
     @PreAuthorize("hasAuthority('user')")
-    public ResponseEntity<CreateResponse> createOrder(Order order) throws MissingFieldsException, EmptyCartException, OrderTimeException {
+    public ResponseEntity<CreateResponse> createOrder(Order order, Long userId) throws MissingFieldsException, EmptyCartException, OrderTimeException, UserNotFoundException {
        if (!order.checkRequiredFields())
```

Comment:
A user can create any user's orders with this scheme. This needs to be changed. I'll meet with you later and show you how I do this.